depot/lib/tram.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Tram do
+  use GenServer
+
+  @moduledoc """
+  Попытка имплементации логики работы трамвая
+  """
+
+  defstruct speed: 0, doors: :closed, accident: false
+
+  def start_link() do
+    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  def init(_init_arg) do
+    {:ok, %{state: :stop, tram: %Tram{}}}
+  end
+
+  def get_state(tram_pid) do
+    GenServer.call(tram_pid, :get_state)
+  end
+
+  def move(tram_pid) do
+    GenServer.call(tram_pid, %{state: :move})
```

Comment:
Лишние обертки обычно только засоряют код
```suggestion
    GenServer.call(tram_pid, :move)
```